OnlineStore/store/src/main/java/Store.java

```diff
@@ -1,96 +1,129 @@
+                    sortedList = products.stream().sorted(Comparator.comparing(Product::getName)).collect(Collectors.toList());
+                } else if (sortOrder.toLowerCase().equals("desc")) {
+                    Comparator<Product> productNameComparatorDesc = Comparator.comparing(Product::getName, (s1, s2) -> {
+                        return s2.compareTo(s1);
+                    });
+                    sortedList = products.stream().sorted(productNameComparatorDesc).collect(Collectors.toList());
+                }
+                break;
+            case "price":
+                if (sortOrder.toLowerCase().equals("asc")) {
+                    sortedList = products.stream().sorted(Comparator.comparing(Product::getPrice)).collect(Collectors.toList());
+                } else if (sortOrder.toLowerCase().equals("desc")) {
+                    Comparator<Product> productPriceComparatorDesc = Comparator.comparing(Product::getPrice, (s1, s2) -> {
+                        return s2.compareTo(s1);
+                    });
+                    sortedList = products.stream().sorted(productPriceComparatorDesc).collect(Collectors.toList());
+                }
+                break;
+            case "rating":
+                if (sortOrder.toLowerCase().equals("asc")) {
+                    sortedList = products.stream().sorted(Comparator.comparing(Product::getRating)).collect(Collectors.toList());
+                } else if (sortOrder.toLowerCase().equals("desc")) {
+                    Comparator<Product> productRatingComparatorDesc = Comparator.comparing(Product::getRating, (s1, s2) -> {
+                        return s2.compareTo(s1);
+                    });
+                    sortedList = products.stream().sorted(productRatingComparatorDesc).collect(Collectors.toList());
+                }
+                break;
+        }
+        return sortedList;
     }
 
-    public ArrayList<Product> sortByRatingAscending(){
-        ProductRatingComparator ratingComparator = new ProductRatingComparator();
-        ArrayList<Product> sortedByPrice = (ArrayList<Product>) products.stream()
-                .sorted(ratingComparator).collect(Collectors.toList());
-        return sortedByPrice;
+    public List<Product> getProductsByCategory(String categoryName){
+        List<Product> productsByCategory = products.stream().filter(p -> categoryName.equals(p.getCategory().getName1()))
+                .collect(Collectors.toList());
+        return productsByCategory;
     }
 
-    public ArrayList<Product> sortByRatingDescending(){
-        ProductRatingComparator ratingComparator = new ProductRatingComparator();
-        ArrayList<Product> sortedByPrice = (ArrayList<Product>) products.stream()
-                .distinct().sorted(ratingComparator).collect(Collectors.toList());
-        return sortedByPrice;
+    /*while in store more then capacity products thread waits*/
+    public List<Product> putProducts(int quantity){
+        locker.lock();
```

Comment:
honestly, i did not get the idea what is happening here, but is seems like overcomplicateded code